### PR TITLE
feat:[PL-32652]: For RBAC CRUD events, reverting to generate v1 events for outbox to maintain compatibility to rollback.

### PR DESCRIPTION
**Pull request by Jimit09Gandhi**
## Describe your changes
As part of https://github.com/harness/harness-core/pull/51759/files we started generating V2 events in outbox and also consuming both v1 and v2 versions. 
This can lead to issues in rollback of previous version understands only v1.
So if there are any v2 events in Outbox, it can fail on deseralization and outbox event processing.
To avoid it for RBAC CRUD events, reverting to generate v1 events for outbox to maintain compatibility and allow for smooth rollback.
To support the change needed had to move HarnessScopeLevel from service to core module. Will revert this change in next release once we start generating V2 events.

Test cases:
Generated Role CRUD operations and checked for outbox event processing.

*******************************************************
We should support changes in below sequence to maintain both forward and backward compatibility of change.

Release-1 -> 
Event Handler handles only v1.
Generates only v1.
**This is existing state in Prod.**

Release-2 -> 
Event Handler handles both v1 and v2.
Generates only v1.

Release-3 -> 
Event Handler handles both v1 and v2.
Generates only v2.
*******************************************************

## Checklist
- [X] I've documented the changes in the PR description.
- [X] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53656)

### CODE REVIEW

**Comment by Jimit09Gandhi** [comment]
trigger uts

**Comment by Jimit09Gandhi** [comment]
trigger codebasehash

**Comment by Jimit09Gandhi** [comment]
trigger ut1 ut2 ut6 ut9

**Comment by Jimit09Gandhi** [comment]
trigger uts

File: access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java [hunk 1]
```diff
@@ -202,11 +208,24 @@ public Role update(Role roleUpdate) {
...
       }
       Role updatedRole = roleDao.update(roleUpdate);
-      outboxService.save(new RoleUpdateEventV2(roleUpdate.getScopeIdentifier(), currentRole, updatedRole));
+      outboxService.save(new RoleUpdateEvent(
+          getAccountId(scopeDTO), RoleMapper.toDTO(currentRole), RoleMapper.toDTO(updatedRole), scopeDTO));
       return updatedRole;
     }));
   }
 
+  private String getAccountId(ScopeDTO scopeDTO) {
```
Comment (ankit-harness) [inline access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java:217]: Optional<String>

File: access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java [hunk 2]
```diff
@@ -202,11 +208,24 @@ public Role update(Role roleUpdate) {
...
+    return Objects.nonNull(scopeDTO) ? scopeDTO.getAccountIdentifier() : null;
+  }
+
+  private ScopeDTO getScopeDTO(Role role) {
```
Comment (ankit-harness) [inline access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java:221]: Optional<ScopeDTO>

**Review by ankit-harness — DISMISSED** [review]
:lgtm:

> Reviewed 32 of 33 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53656#-:-Nfu4n7A5sdUrnS4Fp3J:bnfp4nl)* status: 33 file(s) to review. 1 of 2 approvals obtained.

**Comment by Jimit09Gandhi** [comment]
trigger uts

**Comment by Jimit09Gandhi** [comment]
trigger uts

**Comment by Jimit09Gandhi** [comment]
trigger uts

**Review by Jimit09Gandhi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53656)* status: 34 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java` line 217 at r2](https://reviewable.io/reviews/harness/harness-core/53656#gh-1345518406:-NfuvLujDwqaYxR2PBCx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f1dd87278f3396f38c3370fbe21e496f9f46d369/access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java#L217)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Optional<String>
</blockquote></details>

Done.
___
*[`access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java` line 221 at r2](https://reviewable.io/reviews/harness/harness-core/53656#gh-1345518410:-NfuvLjYBkdeqKzJMckC:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f1dd87278f3396f38c3370fbe21e496f9f46d369/access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/RoleServiceImpl.java#L221)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Optional<ScopeDTO>
</blockquote></details>

Done.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53656#-:-NfuvvDIDAY3xzdurbcU:bnfp4nl)* status: 34 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53656#-:-Nfv0Mcc5PV4AWckR_ev:bnfp4nl)* status: 34 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Pull request merged**